### PR TITLE
Fix Java parser to handle leading annotation

### PR DESCRIPTION
**Pull request by som-snytt**
Noticed at https://github.com/scala/scala3/issues/20026, https://github.com/scala/scala3/issues/20063

### CODE REVIEW

File: test/files/pos/i20026/test.scala [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
+object Test extends App {
+  println {
+    new JTest
```
Comment (lrytz) [inline test/files/pos/i20026/test.scala]: `JFun` and `KTest` are not referenced in the test, maybe they should be to make sure the symbols are forced?

**Review by lrytz — APPROVED** [review]
LGTM!

**Comment by som-snytt** [comment]
on the forward-port, I was too lazy to figure out how to check for `package-info`, then thought maybe it shouldn't bother with that error, since it is javac's job; also, we don't even use the package annotations, so we don't really care.

**Comment by lrytz** [comment]
Moved to 2.13.14 but leaving to @SethTisue to merge or move back.

**Comment by som-snytt** [comment]
I'll tweak per the two comments.

The special weirdness on dotty is because the annotations fix(es) weren't ported.

This tweak might be more niche or lower-priority, since Seth is "risk-sensitive".

**Comment by SethTisue** [comment]
Scala 3 forward-port is https://github.com/scala/scala3/pull/20064

**Pull request merged**